tests/base/test_alarm.py

```diff
@@ -357,12 +386,100 @@ def test_reset(self):
         name = "alarm"
         blank_alarm = watcher.base.Alarm(name=name, callback=None)
         blank_alarm.callback = self.callback
-        for alarm in self.alarm_iter(name=name):
+        for alarm in self.alarm_iter(name=name, callback=self.callback):
             if not alarm.nominal:
                 self.assertNotEqual(alarm, blank_alarm)
             alarm.reset()
             self.assertEqual(alarm, blank_alarm)
 
+    async def test_mute_valid(self):
+        user = "otho"
+        duration = 0.05
+        for severity in AlarmSeverity:
+            if severity == AlarmSeverity.NONE:
+                continue  # invalid value
+            callback = AsyncCallback()
+            for alarm in self.alarm_iter(name=user, callback=callback):
+                alarm.mute(duration=duration, severity=severity, user=user)
+                curr_tai = salobj.tai_from_utc(time.time())
+                await callback.next(timeout=STD_TIMEOUT)
+                self.assertTrue(alarm.muted)
+                self.assertEqual(alarm.muted_by, user)
+                self.assertEqual(alarm.muted_severity, severity)
+                self.assertGreaterEqual(curr_tai + duration, alarm.timestamp_unmute)
+                self.assertAlmostEqual(alarm.timestamp_unmute, curr_tai + duration, places=2)
+                await callback.next(timeout=STD_TIMEOUT)
+                self.assertFalse(alarm.muted)
+                self.assertEqual(alarm.muted_by, "")
+                self.assertEqual(alarm.muted_severity, AlarmSeverity.NONE)
+                self.assertEqual(alarm.timestamp_unmute, 0)
+
+    async def test_mute_invalid(self):
+        good_user = "otho"
+        failed_user = "user associated with invalid mute command"
+        good_duration = 5
+        good_severity = AlarmSeverity.WARNING
+        for alarm in self.alarm_iter(name=good_user, callback=None):
+            for bad_duration, bad_severity in itertools.product((0, -0.01), (AlarmSeverity.NONE, -53)):
+                # check that mute raises ValueError for invalid values
+                # and leaves the alarm state unchanged
+                initial_alarm = copy.copy(alarm)
+                with self.assertRaises(ValueError):
+                    alarm.mute(duration=bad_duration, severity=good_severity, user=failed_user)
+                with self.assertRaises(ValueError):
+                    alarm.mute(duration=good_duration, severity=bad_severity, user=failed_user)
+                with self.assertRaises(ValueError):
+                    alarm.mute(duration=bad_duration, severity=bad_severity, user=failed_user)
+                self.assertEqual(alarm, initial_alarm)
+
+                # make sure failures also leave muted alarm state unchanged
+                alarm.mute(duration=good_duration, severity=good_severity, user=good_user)
```

Comment:
You don't do this check below for *each* of the "bad mutings". I think you've coded `mute` safely enough, but I'd be happier if you did the `assertEqual` after each `assertRaises`.